### ISSUE TITLE
Remove deprecated error codes

### ISSUE THREAD

**Issue by mpg** [ARMmbed/mbedtls#4283]
Remove deprecated error codes, which can be found using `git grep -i 'MBEDTLS_ERR.*deprecated'.

Normally, those codes are already no longer used in the library, programs and test, and only their definition remains, which should be removed. However there are a few exceptions:

- `library/error.c` uses them but it's generated, so it just needs to be re-generated after updating the header files
- `library/psa_crypto.c` uses them in a large `switch` statement in `mbedtls_to_psa_error()`: the corresponding `case`s need to be removed
- `include/mbedtls/error.h` documents the ranges used by each module and might need to be updated

**Comment by TRodziewicz** [comment]
In this task MBEDTLS_ERR_CIPHER_HW_ACCEL_FAILED has been deprecated but it still used in active code in a switch/case as a default: return value. What to use here instead?

file cipher.c, from line 322: 
```     
switch( status )
{
    case PSA_SUCCESS:
        break;
    case PSA_ERROR_INSUFFICIENT_MEMORY:
        return( MBEDTLS_ERR_CIPHER_ALLOC_FAILED );
    case PSA_ERROR_NOT_SUPPORTED:
        return( MBEDTLS_ERR_CIPHER_FEATURE_UNAVAILABLE );
    default:
        return( MBEDTLS_ERR_CIPHER_HW_ACCEL_FAILED ); // <----???
}
```

**Comment by gilles-peskine-arm** [comment]
Use `MBEDTLS_ERR_PLATFORM_HW_ACCEL_FAILED` instead of other `MBEDTLS_ERR_xxx_HW_ACCEL_FAILED`.

**Comment by TRodziewicz** [comment]
This issue has been fixed by #4339 so it can be closed now.
